File: ruby/tests/filetypes_test.rb
################################################################################
#
# filetypes_test.rb
#
# Test the FileTypes class
#
################################################################################

require_relative '../rbsearch/filetypes.rb'
require 'test/unit'

class FileTypesTest < Test::Unit::TestCase
  def setup
    @filetypes = FileTypes.new
  end

  def test_get_filetype_archive_file
    filename = 'archive.zip'
    assert_equal(@filetypes.get_filetype(filename), FileType::ARCHIVE)
  end

  def test_get_filetype_binary_file
    filename = 'binary.exe'
    assert_equal(@filetypes.get_filetype(filename), FileType::BINARY)
  end

  def test_get_filetype_code_file
    filename = 'code.rb'
    assert_equal(@filetypes.get_filetype(filename), FileType::CODE)
  end

  def test_get_filetype_text_file
    filename = 'text.txt'
    assert_equal(@filetypes.get_filetype(filename), FileType::TEXT)
  end

  def test_get_filetype_xml_file
    filename = 'markup.xml'
    assert_equal(@filetypes.get_filetype(filename), FileType::XML)
  end

  def test_get_filetype_searchable_file
    filename = 'config.ini'
    assert_true(@filetypes.searchable_file?(filename))
  end

  def test_get_filetype_unknown_file
    filename = 'unknown.xyz'
    assert_equal(@filetypes.get_filetype(filename), FileType::UNKNOWN)
  end
end

File: python/tests/filetypes_test.py
# -*- coding: utf-8 -*-
################################################################################
#
# filetypes_test.py
#
# class FileTypesTest: testing of FileTypes
#
################################################################################
import sys
import unittest

from pysearch.filetypes import FileType, FileTypes

sys.path.insert(0, '/Users/cary/src/git/xsearch/python')

class FileTypesTest(unittest.TestCase):
    @classmethod
    def setUpClass(self):
        self.filetypes = FileTypes()

    def test_get_filetype_archive_file(self):
        filename = 'archive.zip'
        self.assertEqual(self.filetypes.get_filetype(filename), FileType.Archive)

    def test_get_filetype_binary_file(self):
        filename = 'binary.exe'
        self.assertEqual(self.filetypes.get_filetype(filename), FileType.Binary)

    def test_get_filetype_text_file(self):
        filename = 'text.txt'
        self.assertEqual(self.filetypes.get_filetype(filename), FileType.Text)

    def test_get_filetype_unknown_file(self):
        filename = 'unknown.xyz'
        self.assertEqual(self.filetypes.get_filetype(filename), FileType.Unknown)

if __name__ == '__main__':
    unittest.main()
